tenant/store.go

```diff
@@ -76,84 +79,64 @@ func (tenantStore *tenantStore) listSegments(tenantId uint64) ([]Segment, error)
 }
 
 func (tenantStore *tenantStore) addTenant(tenant *Tenant) error {
+	log.Println("In tenantStore addTenant().")
+
 	var tenants []Tenant
 	tenantStore.DbStore.Db.Find(&tenants)
 
-	//	myId := tenant.Id
+	// Most UUID generators return , or "" if uuid is
+	// invalid thus len != 32 should be rejected.
+	// Create new UUID in case one is not provided.
+	if len(tenant.UUID) != 32 {
```

Comment:
the reasoning was that kubernetes allows namespace but openstack can have multiple namespaces per tenant, thus by allowing this we can map multiple namespaces from kubernetes to a single tenant, for example, namespace in kubernetes is same network namespace which doesn't cross host boundary, but tenants can and thus multiple kubernetes host can have different namespaces which can map onto same tenant id generated here.